code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -273,14 +273,49 @@
 		ITEM_SLOT_LPOCKET, ITEM_SLOT_RPOCKET,\
 		ITEM_SLOT_DEX_STORAGE
 	)
+	var/kicked_over = FALSE
 
-/obj/item/reagent_containers/glass/bucket/wooden
-	name = "wooden bucket"
-	icon_state = "woodbucket"
-	inhand_icon_state = "woodbucket"
-	custom_materials = list(/datum/material/wood = MINERAL_MATERIAL_AMOUNT * 2)
-	armor = list(MELEE = 10, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 0, ACID = 50)
-	resistance_flags = FLAMMABLE
+/obj/item/reagent_containers/glass/bucket/Initialize(mapload)
+	. = ..()
+	update_appearance()
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = .proc/kicking_the_bucket,
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/item/reagent_containers/glass/bucket/proc/bucket_got_kicked()
+	kicked_over = TRUE
+	update_appearance()
+	playsound(src, 'sound/effects/spray2.ogg', 50, TRUE)
+	visible_message(span_danger("\The [src] got kicked!"))
+	chem_splash(loc, null, 3, list(reagents))
+
+/obj/item/reagent_containers/glass/bucket/proc/kicking_the_bucket(datum/source, atom/movable/AM, thrown_at = FALSE)
+	SIGNAL_HANDLER
+	if(!reagents?.total_volume)
+		return
+
+	if(!isturf(loc) || !isliving(AM))
+		return
+	var/mob/living/kicker = AM
+	var/snap = TRUE
+	if(istype(kicker.buckled, /obj/vehicle))
+		var/obj/vehicle/ridden_vehicle = kicker.buckled
+		bucket_got_kicked()
+		ridden_vehicle.visible_message(span_danger("[ridden_vehicle] kicks \the [src]."))
+
+	if(!thrown_at && kicker.movement_type & (FLYING|FLOATING)) //won't kick the bucket if you are flying
+		snap = FALSE
+
+	else if(snap && isanimal(kicker))
+		var/mob/living/simple_animal/simple_kicker = kicker
+		if(simple_kicker.mob_size <= MOB_SIZE_TINY) //don't let small animals kick the bucket
+			snap = FALSE
```

Comment:
snap should not exist, just early return

Reply:
Updated the code to be less bad (hopefully) and removed those leftovers from when I copied beartrap code.